cilium/socket_option_bpf_metadata.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include <asm-generic/socket.h>
+#include <netinet/in.h>
+
+#include <cerrno>
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "envoy/common/pure.h"
+#include "envoy/config/core/v3/socket_option.pb.h"
+#include "envoy/network/address.h"
+#include "envoy/network/socket.h"
+
+#include "source/common/common/logger.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+#include "cilium/network_policy.h"
+#include "cilium/policy_id.h"
+
+namespace Envoy {
+namespace Cilium {
+
+class PolicyResolver {
+public:
+  virtual ~PolicyResolver() = default;
+
+  virtual uint32_t resolvePolicyId(const Network::Address::Ip*) const PURE;
+  virtual const PolicyInstanceConstSharedPtr getPolicy(const std::string&) const PURE;
+};
+
+// Socket Option that holds relevant connection & policy information that can be retrieved
+// by the Cilium network- and HTTP policy filters by using GetBpfMetadataSocketOption.
+class BpfMetadataSocketOption : public Network::Socket::Option,
```

Comment:
maybe worth to rename this to `PolicyMetadataSocketOption` or `PolicyContextSocketOption` - as it's used by the policy filters (network & http)?

Reply:
`CiliumPolicySocketOption` :+1: (prefer to have a consistent suffix `SocketOption`)

Reply:
renamed